src/matrixrtc/MatrixRTCSessionManager.ts

```diff
@@ -98,8 +98,22 @@ export class MatrixRTCSessionManager extends TypedEventEmitter<MatrixRTCSessionM
         return this.roomSessions.get(room.roomId)!;
     }
 
-    private async consumeCallEncryptionEvent(event: MatrixEvent): Promise<void> {
+    private async consumeCallEncryptionEvent(event: MatrixEvent, isRetry = false): Promise<void> {
         await this.client.decryptEventIfNeeded(event);
+        if (event.isDecryptionFailure() && !isRetry) {
+            logger.warn(
+                `Decryption failed for event ${event.getId()}: ${event.decryptionFailureReason} will retry once only`,
+            );
+            // retry after 1 second. After this we give up.
+            setTimeout(() => this.consumeCallEncryptionEvent(event, true), 1000);
+            return;
+        } else if (event.isDecryptionFailure()) {
+            logger.warn(`Decryption failed for event ${event.getId()}: ${event.decryptionFailureReason}`);
+            return;
+        } else if (isRetry) {
+            logger.info(`Decryption succeeded for event ${event.getId()} after retry`);
+        }
+
```

Comment:
I think the following is semantically equivalent while avoiding some redundant branches & an extra call to `event.isDecryptionFailure()`
```suggestion
        if (event.isDecryptionFailure()) {
            if (!isRetry) {
                logger.warn(
                    `Decryption failed for event ${event.getId()}: ${event.decryptionFailureReason} will retry once only`,
                );
                // retry after 1 second. After this we give up.
                setTimeout(() => this.consumeCallEncryptionEvent(event, true), 1000);
            } else {
                logger.warn(`Decryption failed for event ${event.getId()}: ${event.decryptionFailureReason}`);
            }
            return;
        } else if (isRetry) {
            logger.info(`Decryption succeeded for event ${event.getId()} after retry`);
        }

```